src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/PackageReferenceProject.cs

```diff
@@ -218,7 +219,31 @@ internal async Task<ProjectPackages> GetInstalledAndTransitivePackagesAsync(bool
                             targetsList = await GetTargetsListAsync(assetsFilePath, token);
                         }
 
-                        transitiveOrigins = calculatedTransitivePackages.Any() ? ComputeTransitivePackageOrigins(calculatedInstalledPackages, targetsList, token) : new Dictionary<string, TransitiveEntry>();
+                        Dictionary<string, TransitiveEntry> transitiveOriginsCache = new();
+
+                        // Compute Transitive Origins
+                        transitiveOrigins = calculatedTransitivePackages.Any() ? ComputeTransitivePackageOrigins(calculatedInstalledPackages, targetsList, transitiveOriginsCache, token) : transitiveOriginsCache;
+
+                        // If the project has project references, we need to compute transitive origins for their packages
```

Comment:
Wouldn't it be more performant to calculative the transitive origins all at once? 

Right now, we're doing a 2 step and it's not clear to me why it needs to be a 2 step with potentially the same packages.